### PR TITLE
[Fix] Remove single-node validation from jobs clusters

### PR DESCRIPTION
**Pull request by shreyas-goenka**
## Changes
Fixes https://github.com/databricks/cli/issues/1896. Introducing this validation has caused a regression for both DABs and TF customers. This PR removes the validation for job clusters.

## Tests
Unit tests.

### CODE REVIEW

File: clusters/clusters_api.go [hunk 1]
```diff
@@ -434,22 +434,6 @@ type Cluster struct {
 	ClusterMounts    []MountInfo   `json:"cluster_mount_infos,omitempty" tf:"alias:cluster_mount_info"`
 }
 
-// TODO: Remove this once all the resources using clusters are migrated to Go SDK.
-// They would then be using Validate(cluster compute.CreateCluster) defined in resource_cluster.go that is a duplicate of this method but uses Go SDK.
-func (cluster Cluster) Validate() error {
```
Comment (shreyas-goenka) [inline clusters/clusters_api.go:439]: this function is not used anywhere.

**Review by alexott — APPROVED** [review]

**Comment by alexott** [comment]
Do we still need #4206?

**Comment by shreyas-goenka** [comment]
@alexott Yeah we still need https://github.com/databricks/terraform-provider-databricks/pull/4206 since it tackles the issue of policy not working from clusters. I need to update that PR however.

**Comment by tanmay-db** [comment]
> @shreyas-goenka Please update this as well:
> 
> https://github.com/databricks/terraform-provider-databricks/blob/e3b25617d9c9357eccc10fba2eacbd26ac3e7118/clusters/resource_cluster.go#L133-L135
> 
> @tanmay-db Can you confirm if this aligns with the direction you're taking for these resources?

Hi @pietern, yes this would be implemented a bit differently in plugin framework so it should be good. Eventually we will remove the cluster resource in SDKv2 so there would only be one place where validation would happen.

**Pull request merged**